From kristen.c.accardi@intel.com Mon Oct 30 12:08:28 2006
Date: Mon, 30 Oct 2006 13:08:04 -0800
From: Akinobu Mita <akinobu.mita@gmail.com>
To: gregkh@suse.de
Cc: Akinobu Mita <akinobu.mita@gmail.com>, Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Subject: acpiphp: fix missing acpiphp_glue_exit()
Message-Id: <20061030130804.9e58c24c.kristen.c.accardi@intel.com>
Mime-Version: 1.0
Content-Type: text/plain; charset=US-ASCII

From: Akinobu Mita <akinobu.mita@gmail.com>

acpiphp_glue_exit() needs to be called to unwind when no slots found.
(It fixes data corruption when reloading acpiphp driver with no such devices)

Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/acpiphp_core.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/acpiphp_core.c
+++ gregkh-2.6/drivers/pci/hotplug/acpiphp_core.c
@@ -303,8 +303,10 @@ static int __init init_acpi(void)
 	/* read initial number of slots */
 	if (!retval) {
 		num_slots = acpiphp_get_num_slots();
-		if (num_slots == 0)
+		if (num_slots == 0) {
+			acpiphp_glue_exit();
 			retval = -ENODEV;
+		}
 	}
 
 	return retval;
